client/state/plugins/installed/actions.js

```diff
@@ -131,7 +144,11 @@ export function activatePlugin( siteId, plugin ) {
 			pluginId,
 		};
 
-		if ( plugin.active ) {
+		if (
+			isSiteDataOnPluginObject( plugin )
+				? plugin.active
+				: siteId && plugin.sites?.[ siteId ]?.active
```

Comment:
This looks a bit like a code smell - why are we checking if the plugin has `version` or `update` when we're going to be working with the `active` prop? 

Perhaps instead we could pass the prop we're looking for to the function and return true only if that prop is there, and not if some unrelated props are there?

Also, this seems a bit repetitive. Perhaps we can bake in the check inside the function?

Reply:
I changed the approach. Let me know what you think.

Reply:
Thank you, I think it looks great when abstracted 👍